client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -118,23 +121,27 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		const planSlug = getUpgradePlanSlugFromPath( this.props.product, this.props.selectedSite );
-
-		let cartItem,
-			cartMeta;
-
-		if ( planSlug ) {
-			cartItem = getCartItemForPlan( planSlug );
+		const { product, productsList, purchaseId, selectedSite, selectedSiteSlug } = this.props;
+		const meta = product.split( ':' )[ 1 ];
+		const productSlug = getUpgradePlanSlugFromPath( product, selectedSite.jetpack ) || product.split( ':' )[ 0 ];
+
+		let cartItem = getCartItemForPlan( productSlug );
+
+		if ( 'theme' === productSlug ) {
+			cartItem = themeItem( meta );
+		} else if ( 'domain-mapping' === productSlug ) {
+			cartItem = domainMapping( { domain: meta } );
+		} else if ( 'no-ads' === productSlug ) {
```

Comment:
We're introducing some additional product type logic here. Can we add tests for it; also can we verify that this covers all type of products that can be added to the cart? I have the sensation that we might be missing some products that are renewable/upgradeable. Also, should we add all of these cases to testing instructions?